rpc/client.go

```diff
@@ -161,7 +159,7 @@ func (c *Client) connect(opts *retry.Options, context *Context) {
 		// Launch periodic heartbeat.
 		go c.startHeartbeat()
 
-		return retry.Break, nil
+		return retry.Break, err
```

Comment:
Again, this is the success path and we know err to be nil, so we should just return nil.